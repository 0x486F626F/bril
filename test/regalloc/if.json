{
  "functions": [
    {
      "instrs": [
        {
          "dest": "v0",
          "op": "const",
          "type": "int",
          "value": 0
        },
        {
          "dest": "v2",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "dest": "b",
          "op": "const",
          "type": "bool",
          "value": false
        },
        {
          "dest": "vb",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "dest": "vc",
          "op": "const",
          "type": "int",
          "value": 6
        },
        {
          "args": [
            "vb",
            "vc"
          ],
          "dest": "va",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "v0",
            "va"
          ],
          "dest": "vd",
          "op": "sub",
          "type": "int"
        },
        {
          "args": [
            "vc",
            "vd"
          ],
          "dest": "ve",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "b",
            "one",
            "two"
          ],
          "op": "br"
        },
        {
          "label": "one"
        },
        {
          "args": [
            "v2",
            "ve"
          ],
          "dest": "vc",
          "op": "mul",
          "type": "int"
        },
        {
          "args": [
            "end"
          ],
          "op": "jmp"
        },
        {
          "label": "two"
        },
        {
          "args": [
            "vd",
            "ve"
          ],
          "dest": "vb",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "ve",
            "v2"
          ],
          "dest": "ve",
          "op": "sub",
          "type": "int"
        },
        {
          "label": "end"
        },
        {
          "args": [
            "vc",
            "ve"
          ],
          "dest": "vb",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vb"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
