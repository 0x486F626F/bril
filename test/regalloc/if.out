main {
	__r02: int = const 0;
	v2: int = const 2;
	__r03: bool = const false;
	__r01: int = const 5;
	__r04: int = const 6;
	vb: int = id __r01;
	va: int = id __r01;
	__r01: int = id vb;
	__r01: int = add __r01 __r04;
	vd: int = id __r01;
	__r01: int = id va;
	__r01: int = sub __r02 __r01;
	__r02: int = add __r04 __r01;
	br __r03 one two;
one:
	__r04: int = mul v2 __r02;
	jmp end;
two:
	vb: int = id __r01;
	__r01: int = id vd;
	__r01: int = add __r01 __r02;
	__r02: int = sub __r02 v2;
end:
	__r01: int = add __r04 __r02;
	print __r01;
}
